### PR TITLE
Adding a Code of Conduct

### PR DESCRIPTION
**Pull request by JoshData**
Hello all,

This pull request adds a Code of Conduct to this project. It is adapted primarily from the [Contributor Covenant](http://contributor-covenant.org).

I am open to feedback about this draft and what to put in a Code of Conduct.

But _not_ up for discussion is _whether_ to have a Code of Conduct: We will adopt a Code of Conduct.

This isn't in response to anything that has happened in the project so far. I have easily been the most difficult person in this community. But a Code of Conduct signals to would-be community members that we take community seriously and that I am prepared to enforce norms for acceptable behavior when necessary. Tech communities have been historically, and unfortunately too often still continue to be, unwelcoming to women and other groups, and we all have a responsibility to change that.

So please read the draft CoC in this pull request and share your thoughts. I have [cross-posted this on the discussion forum](https://discourse.mailinabox.email/t/adopting-a-code-of-conduct/1388) as well, and I will be watching both places for feedback.

### CODE REVIEW

**Comment by NetOperatorWibby** [comment]
Looks good to me! It's always a great idea to be proactive instead of reactive.

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+The maintainers of Mail-in-a-Box share the dual responsibility of leading by example and enforcing these policies as necessary to maintain an open and welcoming environment.
+
+## Scope
+
+This Code of Conduct applies to all places where Mail-in-a-Box community activity is ocurring, including in on Github, in discussion forums, on Slack, on social media, and in real life. The Code of Conduct applies not only on websites/at events run by the Mail-in-a-Box community (e.g. our Github organization, our Slack team) but also at any other location where the Mail-in-a-Box community is present (e.g. in issues of other Github organizations where Mail-in-a-Box community members are discussing problems related to Mail-in-a-Box, or real-life professional conferences), or whenever a Mail-in-a-Box community member is representing Mail-in-a-Box to the public at large or acting on behalf of Mail-in-a-Box.
```
Comment (yodax) [inline CODE_OF_CONDUCT.md]: > including in on Github

Shouldn't that be "on Github"?

**Comment by yodax** [comment]
Very good initiative! 

I'm (naively maybe) surprised sometimes that these things are needed and I'm often appalled about the horror stories you sometimes hear. 

Perhaps a second contact person would be a good idea. If someone has a complaint/issue regarding @JoshData or wants a second opinion, this might help. Both are unlikely 😄, but I think it shows openness.

**Comment by mar1u5** [comment]
Just a little typo-thing: `GitHub` instead of `Github` 😁

**Comment by guyzmo** [comment]
That's indeed a good idea, @JoshData ! 

I do agree with @yodax for including a a second person and maybe even a third person as contacts regarding any possible abuses…

I've also been told that having *"diversity"* (I really do not like that word, as it emphasize what should not matter…) amongst those contacts should help, as some people can talk more easily about some community issues to non-white-geeky-dudes like most of us are.

Finally, there's something I've heard a lot in hackerspaces as the fundamental community rule: *Be excellent to each other!*, and I think it would fit nice in a link to the code conduct.

my €0.02

**Comment by yodax** [comment]
> I've also been told that having "diversity" .....

Well said, I was thinking about that when I wrote my response. I had a hard time phrasing it, so decided to remove it. But I agree. I don't know if there is any one in the community that fits that profile though. 

I like the be excellent motto. Mine is the don't be an ***. Boils down to the something, but excellent is a far more positive way of phrasing it.

**Comment by JoshData** [comment]
@yodax> Perhaps a second contact person would be a good idea

I thought about that but because of the responsibility entailed in being a CoC point of contact, and the fact that we all don't really know each other very well, I'm not ready at this time to add anyone else, but that could change.

@guyzmo> I've also been told that having "diversity" (I really do not like that word, as it emphasize what should not matter…)

Ah but it _does_ matter!

Thanks everyone. I'm incorporating the other suggestions (two typos and _be excellent_).

**Comment by mikl** [comment]
I'd propose excising this part:

> […]micro-aggressive behavior, such as unconstructive criticism, providing corrections that do not improve the conversation (sometimes referred to as "well actually"s)[…]

Those are very subjective, and likely to have chilling effects on community interaction. Who's the arbiter of whether criticism is constructive, or improves the conversation?

**Comment by JoshData** [comment]
> Who's the arbiter of whether criticism is constructive, or improves the conversation?

Should an arbiter be necessary, that would be me.

Thanks for your comments.

**Pull request merged**